src/Compilers/CSharp/Test/Semantic/Semantics/Utf8StringsLiteralsTests.cs

```diff
@@ -0,0 +1,1790 @@
+                // (10,27): error CS0029: Cannot implicitly convert type 'string' to 'System.Span<byte>'
+                //         Span<byte> span = nullValue;
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "nullValue").WithArguments("string", "System.Span<byte>").WithLocation(10, 27),
+                // (11,43): error CS0029: Cannot implicitly convert type 'string' to 'System.ReadOnlySpan<byte>'
+                //         ReadOnlySpan<byte> readonlySpan = nullValue;
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "nullValue").WithArguments("string", "System.ReadOnlySpan<byte>").WithLocation(11, 43)
+                );
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void NoConversionFromNull_02()
+        {
+            var source = @"
+#pragma warning disable CS0219 // The variable is assigned but its value is never used
+using System;
+class C
+{
+    static void Main()
+    {
+        const string nullValue = null;
+        var array = (byte[])nullValue;
+        var span = (Span<byte>)nullValue;
+        var readonlySpan = (ReadOnlySpan<byte>)nullValue;
+    }
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+
+            // PROTOTYPE(UTF8StringLiterals) : More specific error message?
+            comp.VerifyEmitDiagnostics(
+                // (9,21): error CS0030: Cannot convert type 'string' to 'byte[]'
+                //         var array = (byte[])nullValue;
+                Diagnostic(ErrorCode.ERR_NoExplicitConv, "(byte[])nullValue").WithArguments("string", "byte[]").WithLocation(9, 21),
+                // (10,20): error CS0030: Cannot convert type 'string' to 'System.Span<byte>'
+                //         var span = (Span<byte>)nullValue;
+                Diagnostic(ErrorCode.ERR_NoExplicitConv, "(Span<byte>)nullValue").WithArguments("string", "System.Span<byte>").WithLocation(10, 20),
+                // (11,28): error CS0030: Cannot convert type 'string' to 'System.ReadOnlySpan<byte>'
+                //         var readonlySpan = (ReadOnlySpan<byte>)nullValue;
+                Diagnostic(ErrorCode.ERR_NoExplicitConv, "(ReadOnlySpan<byte>)nullValue").WithArguments("string", "System.ReadOnlySpan<byte>").WithLocation(11, 28)
+                );
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void NoConversionFromType_01()
+        {
+            var source = @"
+#pragma warning disable CS0219 // The variable is assigned but its value is never used
+using System;
+class C
+{
+    static void Main()
+    {
+        string value = ""s"";
+        byte[] array = value;
+        Span<byte> span = value;
+        ReadOnlySpan<byte> readonlySpan = value;
+    }
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+            comp.VerifyEmitDiagnostics(
```

Comment:
Consider also leaving a prototype comment here to give a diagnostic which indicates the requirement of a constant string for the conversion to work.

Reply:
> Consider also leaving a prototype comment here to give a diagnostic which indicates the requirement of a constant string for the conversion to work.

I think the error message here is accurate and sufficiently good.